IQSS/dataverse-frontend
# 198 - File download options

## What this PR does / why we need it:
This PR adds the frontend logic to display the different download options for the files in the Files Table. Only the options under the Download Options header in the Access File Menu.

![image](https://github.com/IQSS/dataverse-frontend/assets/23359572/d60b08b0-8d9c-4d05-a42f-31e86ce31fdd)

## Which issue(s) this PR closes:

- Closes #198 

## Special notes for your reviewer:
I left some TODOs to add support for the Guestbook and the File packages in the future.

## Suggestions on how to test this:
1. Install the necessary dependencies using `npm install`.
5. Build the design system with `cd packages/design-system && npm run build`.
6. Navigate back to the root directory with `cd ../../`.
7. Start Storybook with `npm run storybook`.
8. Visit [the Access File Menu in the Storybook](http://localhost:6006/?path=/story/sections-dataset-page-datasetfiles-filestable-fileactionscell-fileactionbuttons-accessfilemenu--default) to check the download options for tabular and non-tabular files.

## Does this PR introduce a user interface change? If mockups are available, please link/include them here:
### For non tabular files
![image](https://github.com/IQSS/dataverse-frontend/assets/23359572/d60b08b0-8d9c-4d05-a42f-31e86ce31fdd)

### For tabular files
![Screen Shot 2023-10-05 at 2 35 48 PM](https://github.com/IQSS/dataverse-frontend/assets/2364928/d0004ddc-c555-46c6-86fe-e43bb86bdeb9)

## Is there a release notes update needed for this change?:
No

## Additional documentation:
